src/include/storage/index/index.h

```diff
@@ -91,6 +100,18 @@ class Index {
    * @return oid of this indes
    */
   catalog::index_oid_t GetOid() const { return oid_; }
+
+  /**
+   * @return mapping from key oid to projected row offset
+   */
+  const std::unordered_map<catalog::indexkeycol_oid_t, uint32_t> &GetKeyOidToOffsetMap() const {
```

Comment:
I wonder if it makes more sense for the map to be uint16_t so that it matches the input expected of ProjectedRow::AccessX(.) methods.

Would have saved a fair bit of stupid static_casts.